api/cluster.go

```diff
@@ -312,103 +312,136 @@ func (s *Server) peerQuery(ctx context.Context, data cluster.Traceable, name, pa
 // name:         name to be used in logging & tracing
 // path:         path to request on
 func (s *Server) peerQuerySpeculative(ctx context.Context, data cluster.Traceable, name, path string) (map[string]PeerResponse, error) {
-	peerGroups, err := cluster.MembersForSpeculativeQuery()
-	if err != nil {
-		log.Error(3, "HTTP peerQuery unable to get peers, %s", err)
-		return nil, err
-	}
-	log.Debug("HTTP %s across %d instances", name, len(peerGroups)-1)
+	result := make(map[string]PeerResponse)
 
-	reqCtx, cancel := context.WithCancel(ctx)
-	defer cancel()
+	responseChan, errorChan := s.peerQuerySpeculativeChan(ctx, data, name, path)
 
-	originalPeers := make(map[string]struct{}, len(peerGroups))
-	receivedResponses := make(map[int32]struct{}, len(peerGroups))
+	for resp := range responseChan {
+		result[resp.peer.GetName()] = resp
+	}
 
-	responses := make(chan struct {
-		shardGroup int32
-		data       PeerResponse
-		err        error
-	}, 1)
+	err := <-errorChan
+	return result, err
+}
 
-	askPeer := func(shardGroup int32, peer cluster.Node) {
-		log.Debug("HTTP Render querying %s%s", peer.GetName(), path)
-		buf, err := peer.Post(reqCtx, name, path, data)
+// peerQuerySpeculativeChan takes a request and the path to request it on, then fans it out
+// across the cluster. If any peer fails requests to other peers are aborted. If enough
```

Comment:
I notice `peerQuerySpeculative` description includes "...except to the local peer...", whereas this one does not.
i think the former is wrong